src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/RestAnomalyDetectorJobAction.java

```diff
@@ -80,14 +83,33 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
         String detectorId = request.param(DETECTOR_ID);
         long seqNo = request.paramAsLong(IF_SEQ_NO, SequenceNumbers.UNASSIGNED_SEQ_NO);
         long primaryTerm = request.paramAsLong(IF_PRIMARY_TERM, SequenceNumbers.UNASSIGNED_PRIMARY_TERM);
+        boolean historical = request.paramAsBoolean("historical", false);
         String rawPath = request.rawPath();
+        DetectionDateRange detectionDateRange = getDetectionDateRange(request);
 
-        AnomalyDetectorJobRequest anomalyDetectorJobRequest = new AnomalyDetectorJobRequest(detectorId, seqNo, primaryTerm, rawPath);
+        AnomalyDetectorJobRequest anomalyDetectorJobRequest = new AnomalyDetectorJobRequest(
+            detectorId,
+            detectionDateRange,
+            historical,
+            seqNo,
+            primaryTerm,
+            rawPath
+        );
 
         return channel -> client
             .execute(AnomalyDetectorJobAction.INSTANCE, anomalyDetectorJobRequest, new RestToXContentListener<>(channel));
     }
 
+    private DetectionDateRange getDetectionDateRange(RestRequest request) throws IOException {
```

Comment:
`RestAnomalyDetectorJobAction` is the user interface which will accept the REST request. This is to parse detection date range from the request. Will change it to `parseDetectionDateRange` to avoid the confusion.